batch/batch/instance_config.py

```diff
@@ -94,34 +94,33 @@ def _cost_per_hour_from_resources(resource_rates: Dict[str, float], resources: L
             cost_per_msec += quantity * rate_unit_msec
         return cost_per_msec * 1000 * 60 * 60
 
-    def cost_per_hour(
+    # The price a user pays to rent a portion of this instance
+    def price_per_hour(
         self,
         resource_rates: Dict[str, float],
         cpu_in_mcpu: int,
         memory_in_bytes: int,
         storage_in_gb: int,
     ) -> float:
-        resources = self.quantified_resources(cpu_in_mcpu, memory_in_bytes, storage_in_gb)
-        return InstanceConfig._cost_per_hour_from_resources(resource_rates, resources)
+        resources_billed_to_user = self.quantified_resources(cpu_in_mcpu, memory_in_bytes, storage_in_gb)
+        return InstanceConfig._cost_per_hour_from_resources(resource_rates, resources_billed_to_user)
 
-    def cost_per_hour_from_cores(
+    def entire_instance_price_per_hour(
         self,
         resource_rates: Dict[str, float],
-        utilized_cores_mcpu: int,
     ) -> float:
-        assert 0 <= utilized_cores_mcpu <= self.cores * 1000
-        if self.job_private:
-            assert utilized_cores_mcpu == self.cores * 1000
-            memory_in_bytes = self.instance_memory()
-        else:
-            memory_in_bytes = self.cores_mcpu_to_memory_bytes(utilized_cores_mcpu)
```

Comment:
I think now the function cores_mcpu_to_memory_bytes can be deleted from the Instance Config classes (this one and the azure and gcp implementations)

Reply:
Great catch! addressed